frontend/src/views/consumer/event-listing/EventBooking.jsx

```diff
@@ -0,0 +1,283 @@
+    internalState = observable({
+      customerGroup: undefined,
+      tickets: [
+        {
+          id: 'first',
+          label: this.props.event.ticketCatalog[0].ticketDescription,
+          value: this.props.event.ticketCatalog[0].id,
+          amount: 0,
+        },
+      ],
+      phoneNumber: '',
+      name: '',
+      email: '',
+      classRoom: '',
+      school: '',
+    });
+    getTotalPrice = () => {
+      return this.internalState.tickets.reduce((acc, curr) => {
+        const targetCatalog = this.props.event.ticketCatalog.find(
+          elem => elem.id == curr.value,
+        );
+        if (targetCatalog) {
+          const singlePrice = targetCatalog.price;
+          const totalPriceForThisCatalog = singlePrice * curr.amount;
+          acc += totalPriceForThisCatalog;
+        }
+        return acc;
+      }, 0);
+    };
+    render() {
+      const { event } = this.props;
+      const { internalState } = this;
+
+      const isPrivateCustomer = internalState.customerGroup == 'private';
+      const isGroupConductorCustomer = internalState.customerGroup == 'group';
+      const totalCost = this.getTotalPrice();
+      const submittable = isGroupConductorCustomer
+        ? internalState.school &&
+          internalState.classRoom &&
+          isValidNumber(internalState.phoneNumber, 'FI')
+        : internalState.name && isValidNumber(internalState.phoneNumber);
+      return (
+        <Wrapper bgColor={event.themeColor}>
+          <Typography type="title">Varaa/osta lippu</Typography>
+          <FormListItem title="1 - Olen">
+            <ButtonGroup style={{ display: 'flex' }}>
+              <CustomerGroupButton
+                backgroundColor={
+                  isGroupConductorCustomer ? event.themeColor : 'white'
+                }
+                onClick={e => (internalState.customerGroup = 'group')}
+                onTouchEnd={e => (internalState.customerGroup = 'group')}
+              >
+                Ryhmänohjaaja
+                {isGroupConductorCustomer && <Icon type="check" />}
+              </CustomerGroupButton>
+              <CustomerGroupButton
+                onClick={e => (internalState.customerGroup = 'private')}
+                onTouchEnd={e => (internalState.customerGroup = 'private')}
+                backgroundColor={isPrivateCustomer ? event.themeColor : 'white'}
+              >
+                Yksityishenkilö
+                {isPrivateCustomer && <Icon type="check" />}
+              </CustomerGroupButton>
+            </ButtonGroup>
+          </FormListItem>
+          <FormListItem
+            title="2 - Lippujen määrä"
+            disabled={!internalState.customerGroup}
+          >
+            <TicketInputSet
+              ticketCatalog={event.ticketCatalog}
+              tickets={internalState.tickets}
+            />
+            <Button
+              disabled={
+                internalState.tickets.length >= event.ticketCatalog.length
```

Comment:
This business logic could also be in a method instead of inlined inside render..